.github/workflows/core-smart-contract-deploy.yml

```
@@ -0,0 +1,116 @@
+name: Smart Contract Deploy
+
+on:
+  workflow_dispatch: # Manual trigger
+    inputs:
+      network:
+        description: 'Network'
+        required: true
+        type: choice
+        options:
+          - hardhat
+          - avalancheFujiTestnet
+          - arbitrumSepolia
+          - arbitrum
+          - bellecour
+        default: 'hardhat'
+      environment:
+        description: 'Environment'
+        required: true
+        type: choice
+        options:
+          - dev
+          - prod
+        default: 'dev'
+
+concurrency:
+  group: ${{ github.ref }}-core-smart-contract-ci
+  cancel-in-progress: true
+
+jobs:
+  build-and-test:
+    uses: ./.github/workflows/core-smart-contract-ci.yml
+    with:
+      node-version: 20
+
+  deploy:
+    needs: build-and-test
+    runs-on: ubuntu-latest
+    env:
+      CI: true
+    permissions:
+      contents: write # Required to commit deployment files.
+    environment: ${{ inputs.network }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+          cache-dependency-path: 'packages/smart-contract'
+
+      - name: Install dependencies
+        working-directory: packages/smart-contract
+        run: npm ci
+
+      - name: Validate deployment environment and prepare variables
+        if: inputs.network != 'hardhat'
+        run: |
+          NETWORK="${{ inputs.network }}"
```

```suggestion
         |
    NETWORK="${{ inputs.network }}"
    ENVIRONMENT="${{ inputs.environment }}"
    
    case "$NETWORK" in
      arbitrum|bellecour)
        if [ "$ENVIRONMENT" = "dev" ]; then
          echo "Error: Cannot deploy to mainnet ($NETWORK) with dev environment"
          exit 1
        fi
        echo "IS_MAINNET=true" >> $GITHUB_ENV
        ;;
      *)
        echo "IS_MAINNET=false" >> $GITHUB_ENV
        ;;
    esac
```
may be there is a way to be improve it using case 🤔

---

[fix: improve mainnet deployment validation in CI workflow](https://github.com/iExecBlockchainComputing/dataprotector-sdk/pull/468/commits/4ffd237294ccfa1b591831a04abbe45a72770e69)